### PR TITLE
✨ v0.3.x: block clusterctl init and move on Kubernetes v1.22

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
See issue for details.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partially implements #4966 (1., only clusterctl in this PR)

### CODE REVIEW

File: cmd/clusterctl/client/cluster/client.go [hunk 1]
```diff
@@ -33,6 +33,9 @@ import (
 
 const (
 	minimumKubernetesVersion = "v1.16.0"
+	// firstUnsupportedKubernetesVersion is the first version we don't support as management cluster.
+	// We cannot support v1.22.0 as it's not compatible with our controller-runtime version and WebhookConfigurations v1beta1.
+	firstUnsupportedKubernetesVersion = "v1.22.0"
```
Comment (sbueringer) [inline cmd/clusterctl/client/cluster/client.go:38]: It was easier to verify it that way. I guess i could also check serverVersion <= v1.21.99 (or something similar)

File: cmd/clusterctl/client/cluster/client.go [hunk 2]
```diff
@@ -228,6 +231,9 @@ type Proxy interface {
 	// ValidateKubernetesVersion returns an error if management cluster version less than minimumKubernetesVersion
 	ValidateKubernetesVersion() error
 
+	// ValidateKubernetesMaxVersion returns an error if management cluster version is greater than or equal to firstUnsupportedKubernetesVersion
```
Comment (sbueringer) [inline cmd/clusterctl/client/cluster/client.go:234]: I'm not sure if it's the best idea to add an additional func. The problem is that `ValidateKubernetesVersion()` is currently only used to check if CRDs can be deployed and it wouldn't make sense to verify serverVersion < v1.22.0 in this case.

File: cmd/clusterctl/client/move.go [hunk 3]
```diff
@@ -62,6 +62,10 @@ func (c *clusterctlClient) Move(options MoveOptions) error {
 		if err != nil {
 			return err
 		}
+		// TODO: do we also want to check this in the dry-run case?
```
Comment (sbueringer) [inline cmd/clusterctl/client/move.go:65]: As noted: do we also want to do this check if DryRun == true?

**Comment by sbueringer** [comment]
@fabriziopandini I opened this PR to start a discussion on how to implement this. I added some conversations above with open questions from my side. Also totally fine if this should be implemented completely different :)

File: cmd/clusterctl/client/cluster/proxy.go [hunk 4]
```diff
@@ -100,6 +100,31 @@ func (k *proxy) ValidateKubernetesVersion() error {
 	return nil
 }
 
+func (k *proxy) ValidateKubernetesMaxVersion() error {
```
Comment (fabriziopandini) [inline cmd/clusterctl/client/cluster/proxy.go:103]: Q, have you considered to implement this check into `ValidateKubernetesVersion`, so we get for free the check to be executed in all the commands where we run `EnsureCustomResourceDefinitions` as one of the first operations...

Comment (sbueringer) [reply cmd/clusterctl/client/cluster/proxy.go:103]: See: https://github.com/kubernetes-sigs/cluster-api/pull/4984#discussion_r673707349

If you think it's okay, I can merge the funcs

Comment (sbueringer) [reply cmd/clusterctl/client/cluster/proxy.go:103]: /cc @fabriziopandini  (+dumb question: does it make a difference if I mention someone or not? :)

Comment (sbueringer) [reply cmd/clusterctl/client/cluster/proxy.go:103]: As discussed let's keep it explicit

File: cmd/clusterctl/client/move.go (hunk 3 above)
Comment (sbueringer) [reply cmd/clusterctl/client/move.go]: As discussed, no we don't

**Comment by sbueringer** [comment]
@fabriziopandini PTAL :)

**Comment by sbueringer** [comment]
@fabriziopandini fyi: I renamed the 1.22 constant to `managementClusterVersionCeiling` (as Vince suggested) to keep it sync with the one which implements the block in KCP

**Review by vincepri — APPROVED** [review]
/approve

**Comment by sbueringer** [comment]
@fabriziopandini I think I just need another lgtm from you for merge (only changed the var name since the last one)

**Pull request merged**